src/main/scala/is/hail/utils/SummaryCombiner.scala

```diff
@@ -39,28 +41,49 @@ class SummaryCombiner[T](f: Iterable[T] => Int) extends Serializable {
     this
   }
 
-  def merge(data: (Variant, (Annotation, Iterable[T]))): SummaryCombiner[T] = {
+  def merge(rv: RegionValue): SummaryCombiner = {
+    require(rowType.fieldType(1).isInstanceOf[TVariant])
```

Comment:
This is unnecessary since the next line will immediately fail with a class cast exception if it isn't a `TVariant`.

Reply:
I think the readability of this code would be greatly improved by a `VariantView`.  Luckily @danking already has a start on that: https://github.com/hail-is/hail/pull/2428.  Can you coordinate with him to extend it to handle this use case?